client/session.go

```diff
@@ -876,9 +923,16 @@ func (s *session) streamBlocksMetadataFromPeer(
 					continue
 				}
 
+				var pChecksum *uint32
+				if b.Checksum != nil {
```

Comment:
What are the use cases where b.Checksum == nil? Are we ok with creating a blockMetadata with checksum of nil pointer?